include/beman/dump/dump.hpp

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef BEMAN_DUMP_DUMP_HPP_INCLUDED
+#define BEMAN_DUMP_DUMP_HPP_INCLUDED
+
+#include <array>
+#include <print>
+#include <string_view>
+
+namespace beman::dump {
+    namespace detail {
+        template <std::size_t N>
+        constexpr auto format_string = []{
+            std::array<char, N * 3 - 1> result{};
+
+            auto ptr = &result[0];
+            for (auto i = 0uz; i != N; ++i)
+            {
+                if (i > 0)
+                    *ptr++ = ' ';
+                *ptr++ = '{';
+                *ptr++ = '}';
+            }
+
+            return result;
+        }();
+
+        template <>
+        inline constexpr std::array<char, 0> format_string<0>;
+    }
+
+    template <typename ... Args>
+    void dump(Args&&... args)
```

Comment:
This function needs documentation.

Reply:
temporary marked a TODO comment, prioritizing merging this commit first.